samza-core/src/main/java/org/apache/samza/zk/ZkKeyBuilder.java

```diff
@@ -88,5 +88,7 @@ public String getJobModelPath(String jobModelVersion) {
     return String.format("%s/%s", getJobModelPathPrefix(), jobModelVersion);
   }
 
-
+  public String getJobModelVersionBarrierPrefix() {
+    return String.format("/%s/versionBarriers", pathPrefix);
```

Comment:
Most of the get*path*() methods use pathPrefix at the start, that's why they use '/' as the first character.
getJobModelVersion() uses getJobModelPathPrefix(), which already includes the '/' at the beginning.

I added a test anyway.